lxd/device/disk.go

```diff
@@ -1235,7 +1245,8 @@ func (d *disk) startVM() (*deviceConfig.RunConfig, error) {
 			// If the source being added is a directory or cephfs share, then we will use the lxd-agent
 			// directory sharing feature to mount the directory inside the VM, and as such we need to
 			// indicate to the VM the target path to mount to.
-			if shared.IsDir(mount.DevPath) || d.sourceIsCephFs() {
```

Comment:
@MggMuggins not a blocker for this PR, but please can you check that the removal of the IsDir check here isn't changing the behaviour in the scenario where a non-dir path is used as disk source for VMs?

Reply:
hrm on 2nd thoughts this does look to be a problem, as i cant see where the differentiation between a dir and non-dir is taking place anymore.

Reply:
Agreed; I've reintroduced the check